README.md

```diff
@@ -2,73 +2,250 @@
-- [Contributing](#contributing)
 - [Credits](#credits)
 
-<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+## Installation and Usage
+
+### Before using the library
+
+The goal of this library is to provide very easy integration of a search bar into your own documentation. To make that possible, you need to scrape your website content in advance, and index it in a MeiliSearch instance.
+
+Luckily, we provide all the tools that you need, and can help you through the whole process, if you follow [this guide](https://docs.meilisearch.com/resources/howtos/search_bar_for_docs.html) 🚀
+
+As a first introduction, you might only want to test this library without connecting it to your website.<br>
+You can do it by running the playground provided in this repository:
+
+```bash
+$ yarn install
+$ yarn serve
+```
+
+The data comes from MeiliSearch documentation.<br>
+Type `create an indxe` to live the MeiliSearch experience with the [typo tolerance](https://docs.meilisearch.com/guides/advanced_guides/typotolerance.html).
+
+### Basic Usage
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <link
+      rel="stylesheet"
+      href="https://cdn.jsdelivr.net/npm/docs-searchbar.js@{version}/dist/cdn/docs-searchbar.min.css"
+    />
+  </head>
+
+  <body>
+    <input type="search" id="search-bar-input" />
+    <script src="https://cdn.jsdelivr.net/npm/docs-searchbar.js@{version}/dist/cdn/docs-searchbar.min.js"></script>
+    <script>
+      docsSearchBar({
+        hostUrl: "https://mymeilisearch.com",
+        apiKey: "XXX",
+        indexUid: "docs",
+        inputSelector: "#search-bar-input",
+        debug: true // Set debug to true if you want to inspect the dropdown
+      });
+    </script>
+  </body>
+</html>
+```
+
+The `hostUrl` and the `apiKey` fields are the credentials of your MeiliSearch instance.<br>
+`indexUid` is the index identifier into the scraper have pushed your website contents.<br>
+`inputSelector` is the CSS identifier of the HTML search input.
+
+_You haven't scraped your documentation contents yet? Check out [this tutorial](https://docs.meilisearch.com/resources/howtos/search_bar_for_docs.html)._
+
+**WARNING: We recommend providing the MeiliSearch public key, which is enough to perform searches.<br>
+Read more about [MeiliSearch authentication](https://docs.meilisearch.com/guides/advanced_guides/authentication.html).**
+
+### Customization
+
+The default behavior of this library fits perfectly for a documentation search bar, but you might need some customizations.
+
+- [Optional parameters](#optional-parameters-) (when calling `docsSearchBar` method)
+- [Styling](#styling-) (with CSS)
+
+#### Optional parameters <!-- omit in toc -->
+
+When calling the `docsSearchBar` method, you can add optional fields:
+
+##### `queryHook` <!-- omit in toc -->
+
+This method is called on every keystroke to transform the typed keywords before querying MeiliSearch. By default, it does not do anything.
```

Comment:
Given the sentence above: 
> When calling the `docsSearchBar` method, you can add optional fields:

I don't understand how `queyHook` is a field and which value I should give him. Do you mean something like this?
```suggestion
queryHook takes a callback function as value. This function will be called on every keystroke to transform the typed keywords before querying MeiliSearch. By default, it does not do anything.
```

The same comment is given to all the other subtitles.